openlibrary/catalog/utils/__init__.py

```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-import re, web, cgitb
+import re, sys, web, cgitb
```

Comment:
Great catching all these issues!  I think the fix here is to handle the exception without sending emails though, that legacy code is not used and is why the missing sys has not caused any problems. I think `catch_and_mail_exception` is not called from anywhere.

Reply:
The trick with _undefined names_ is that some other module might have already imported sys so there isn’t a problem on a particular run but the program order is changed then that _accidental_ good luck no longer holds.

The fixes to #1314 are going to touch a _lot_ of files so I believe the early PRs should be simple fixes that are easy to review and commit.  PRs that change code flow or code intention are going to be harder to test and take longer to review.  Therefore, can you please create a _seperate_ issue on the dead code elimination so that it can be attacked in a separate PR.  I know that that refactoring is a great idea that should not be lost/forgotten but I do not feel knowledgeable on this codebase to do it in this PR.

Reply:
#1453

Reply:
#1453